README.md

```diff
@@ -0,0 +1,767 @@
+
+## Time Bounds
+
+A UCAN's time bounds MUST NOT be considered valid if the current system time is before the `nbf` field or after the `exp` field. This is called the "validity period." Proofs in a chain MAY have different validity periods, but MUST all be valid at execution-time. This has the effect of making a delegation chain valid between the latest `nbf` and earliest `exp`.
+
+``` js
+// Pseudocode
+
+const ensureTime = (delegationChain, now) => {
+  delegationChain.forEach((ucan) => {
+    if (!!ucan.nbf && now < can.nbf) {
+      throw new Error(`Delegation is not yet valid, but will become valid at ${ucan.nbf}`)
+    }
+
+    if (ucan.exp !== null && now > ucan.exp) {
+      throw new Error(`Delegation expired at ${ucan.exp}`)
+    }
+  })
+}
+```
+
+## Principal Alignment
+
+In delegation, the `aud` field of every proof MUST match the `iss` field of the UCAN being delegated to. This alignment MUST form a chain back to the Subject for each resource. 
+
+This calculation MUST NOT take into account [DID fragment]s. If present, fragments are only intended to clarify which of a DID's keys was used to sign a particular UCAN, not to limit which specific key is delegated between. Use `did:key` if delegation to a specific key is desired.
+
+``` mermaid
+flowchart RL
+    invoker((&nbsp&nbsp&nbsp&nbspDan&nbsp&nbsp&nbsp&nbsp))
+    subject((&nbsp&nbsp&nbsp&nbspAlice&nbsp&nbsp&nbsp&nbsp))
+
+    subject -- controls --> resource[(Storage)]
+    rootCap -- references --> resource
+
+    subgraph Delegations
+        subgraph root [Root UCAN]
+            subgraph rooting [Root Issuer]
+                rootIss(iss: Alice)
+                rootSub(sub: Alice)
+            end
+
+            rootCap("cap: (Storage, crud/*)")
+            rootAud(aud: Bob)
+        end
+
+        subgraph del1 [Delegated UCAN]
+            del1Iss(iss: Bob) --> rootAud
+            del1Sub(sub: Alice)
+            del1Aud(aud: Carol)
+            del1Cap("cap: (Storage, crud/*)") --> rootCap
+
+
+            del1Sub --> rootSub
+        end
+
+        subgraph del2 [Delegated UCAN]
+            del2Iss(iss: Carol) --> del1Aud
+            del2Sub(sub: Alice)
+            del2Aud(aud: Dan)
+            del2Cap("cap: (Storage, crud/*)") --> del1Cap
+
+            del2Sub --> del1Sub
+        end
+    end
+
+     subgraph inv [Invocation]
+        invIss(iss: Dan)
+        args("args: [Storage, crud/update, (key, value)]")
+        invSub(sub: Alice)
+        prf("proofs")
+    end
+
+    invIss --> del2Aud
+    invoker --> invIss
+    args --> del2Cap
+    invSub --> del2Sub
+    rootIss --> subject
+    rootSub --> subject
+    prf --> Delegations
+```
+
+### Recipient Validation
+
+An agent executing a capability MUST verify that the outermost `aud` field _matches its own DID._ The associated ability MUST NOT be performed if they do not match. Recipient validation is REQUIRED to prevent the misuse of UCANs in an unintended context.
```

Comment:
Does this belong in the delegation spec ? Perhaps it should be moved to invocation one instead ? Or maybe it should be reframed so it's not about the execution but rather about received delegation ?